app/dojo.py

```diff
@@ -248,3 +251,113 @@ def print_unallocated(self,filename=None):
 			txt_file.write(output)
 			txt_file.close()
 			return ("\nData has been successfully saved to {}.txt\n" .format(filename))
+
+	def reallocate_person(self, emailaddress, new_roomname):
+		"""This method reallocates a person using their unique identifier,
+		in this case their email address, to a the specified new room
+
+		"""
+		new_room = None
+		person_reallocating = None
+		
+		#check if a person with the email address exists
+		for member in itertools.chain(self.all_staff,self.all_fellows):
+			if member.email == emailaddress:
+				person_reallocating = member
+
```

Comment:
Make use of list comprehensions instead of for loops. As you can see, for loops are pushing you to the right . example:
`if email address in [person.email for person in (self.all_staff + self.all_fellows)] `
               `person_reallocating = member`
             `